### PR TITLE
Upgrade sequelize from v5 to v6

### PR DESCRIPTION
**Pull request by vladimyr**
Upgrading `sequelize` (and `sequlieze-cli`) to latest version, see [upgrade docs](https://sequelize.org/master/manual/upgrade-to-v6.html) 📘

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
 const capitalize = require('to-case').capital;
 const contentPluginRegistry = require('./shared/content-plugins');
 const pkg = require('../package.json');
+const Promise = require('bluebird');
 const { promisify } = require('util');
-const sequelize = require('sequelize');
 
 // NOTE: This needs to be done before db models get loaded!
-if (process.env.NODE_ENV !== 'production') {
-  sequelize.Promise.config({ longStackTraces: true });
```
Comment (vladimyr) [inline server/index.js:14]: `Sequelize.Promise` is deprecated.

File: server/shared/database/config.js [hunk 2]
```diff
@@ -6,7 +6,6 @@ const isProduction = process.env.NODE_ENV === 'production';
 
 module.exports = {
   ...readConfig(),
-  operatorsAliases: false,
```
Comment (vladimyr) [inline server/shared/database/config.js:9]: This was deprecated back in v5 ([source](https://bit.ly/3ee5PnD)).

File: server/shared/database/helpers.js [hunk 3]
```diff
@@ -7,17 +7,14 @@ const inRange = require('lodash/inRange');
 const last = require('lodash/last');
 const mapValues = require('lodash/mapValues');
 
-// eslint-disable-next-line no-extra-parens
-const AsyncFunction = (async function () {}).constructor;
-
-const isAsyncFunction = arg => arg instanceof AsyncFunction;
```
Comment (vladimyr) [inline server/shared/database/helpers.js:13]: Now we wrap all methods, not only those declared as `async`.

File: server/shared/database/migrations/20181115140903-course-user-relationship.js [hunk 4]
```diff
@@ -37,11 +37,11 @@ module.exports = {
...
-      return queryInterface.addConstraint(
-        TABLE_NAME,
-        ['course_id', 'user_id'],
-        { type: 'primary key', name: 'course_user_pkey' }
-      );
+      return queryInterface.addConstraint(TABLE_NAME, {
+        name: 'course_user_pkey',
+        type: 'primary key',
+        fields: ['course_id', 'user_id']
+      });
```
Comment (vladimyr) [inline server/shared/database/migrations/20181115140903-course-user-relationship.js:44]: `QueryInterface.addContraint` signature has been updated ([source](https://bit.ly/3gIz183)).

File: server/shared/database/migrations/20200130124252-create-repository-tag.js [hunk 5]
```diff
@@ -17,11 +17,11 @@ exports.up = (queryInterface, Sequelize) => {
...
-    return queryInterface.addConstraint(
-      TABLE_NAME,
-      ['repository_id', 'tag_id'],
-      { type: 'primary key', name: 'repository_tag_pkey' }
-    );
+    return queryInterface.addConstraint(TABLE_NAME, {
+      name: 'repository_tag_pkey',
+      type: 'primary key',
+      fields: ['repository_id', 'tag_id']
+    });
```
Comment (vladimyr) [inline server/shared/database/migrations/20200130124252-create-repository-tag.js:24]: See above ⬆️

File: server/shared/database/helpers.js [hunk 6]
```diff
@@ -82,17 +79,33 @@ function where(attribute, logic, options = {}) {
...
-  });
+function wrapMethods(Model, Promise) {
+  let Ctor = Model;
+  do {
+    const methods = getMethods(Ctor.prototype);
+    const staticMethods = getMethods(Ctor);
+    [...methods, ...staticMethods].forEach(method => wrapMethod(method, Promise));
+    Ctor = Object.getPrototypeOf(Ctor);
+  } while (Ctor !== Sequelize.Model && Ctor !== Function.prototype);
+  return Model;
```
Comment (vladimyr) [inline server/shared/database/helpers.js:90]: /cc @abasic

Comment (underscope) [reply server/shared/database/helpers.js:90]: 🤡 🏋🏻‍♀️

**Pull request merged**